### ISSUE TITLE
Arpeggio Note Release bug (Zyn/VST)

### ISSUE THREAD

**Issue by musikBear** [LMMS/lmms#320]
win32 0.9.90 p.r.
bug
If arpeggio is turned ON (page FUNK | ON) during play in a VST based instrument and in zynaddsubfx, the note will hang. Since zyn is integrated, and dont have a panic button, this is a bug

**Comment by tresf** [comment]
Closing due to lack of response from OP.  Please reopen if you still experience this bug. :beer:

**Comment by musikBear** [comment]
Sadly the bug exists
[ZASFX_HangNoteProblem.zip](https://github.com/LMMS/lmms/files/3666549/ZASFX_HangNoteProblem.zip)

Recreate:
It is the note E3 that creates the hang. The arpeggio caused the issue, but arpeggio -replay is not mandatory to invoke the fail.
Just press E3, and that note hangs
Arpeggio' relation is only by chance: E3 is 'pressed' in the arpeggio sequence, and that the initiate the hang. 
The Hang is terminated if:
* Show-Gui button is pressed
* Panic-button inside xasfx is pressed
The Hang is however not fixed. It returns if E3 is re-pressed. 
I have tested on win1.2.0 and a Master
Both with the bug, but in Master, only after **2**. replay. To me that could indicate some kind of un-precise arpeggio termination signal for the note, but why then is the bug present as soon as E3 is pressed?

**Comment by DomClark** [comment]
This should have been fixed in #4908. For arpeggiated notes, the base note doesn't send MIDI messages itself, only the child notes. However, the code checked whether the arpeggiator was enabled separately for note-on and note-off. Turning the arpeggiator on while the note played meant that the note-on message was sent, but not note-off. Since #4908 this condition is only checked at note-on, and a member variable of `NotePlayHandle` is used to remember whether a note-off is required or not.

@musikBear Unfortunately, I can't reproduce the issue with the file you provided. Is it triggered by turning the arpeggiator on while the note is playing? If not, it's a different bug and you should open a new issue for it. Can you also let us know the full names of the versions you tested this on (including any "rc" parts or final numbers after a dot)?

**Comment by musikBear** [comment]
> I can't reproduce the issue with the file you provided

That is strange! 

> Is it triggered by turning the arpeggiator on while the note is playing?

It is, but that is only because **E3** is played BY the arp. The hang will initiate IF E3 is played, nomatter if it is pressed manually, OR played by the arp
So that make me think, that since YOU cant initiate the bug if you press E3, then it is my zasfx that 'keeps something' in a ? cache..?
Does that exist in the zasfx code???
Take a look at this
https://youtu.be/Jx7jCNv4lAI
*This* is made with the *uploaded file*. I have downloaded the file from *here*, to make sure both you and i had the exact same file!
Could you retry and replicate exactly what i do on that video?
If the result is different, then ?? it is a zasfx 'cache' thing.. ?

The RC i have is : 1.2.0RC7.263

**Comment by DomClark** [comment]
I've managed to reproduce this in 1.2.0 using a MIDI keyboard. I can't trigger it reliably, but so far it's only E3 in this project that's got stuck. Since it can be triggered without the arpeggiator, however, I don't think it's related to this issue. Can you open a new issue for this?

**Comment by musikBear** [comment]
> I've managed to reproduce this in 1.2.0 using a MIDI keyboard

Ok i did it with a PC-kb, but it is good that you could recreate, because now we know that this is not local inside zasfx
> Can you open a new issue for this?

Done its 'Hang-note in zasfx' #5220
